iphone/Classes/TiUIiOSApplicationShortcutsProxy.m

```diff
@@ -173,19 +204,44 @@ -(void)addShortcutItem:(id)args
 
 -(UIApplicationShortcutIcon*)findIcon:(id)value
 {
+    if ([value isKindOfClass:[UIApplicationShortcutIcon class]]) {
+        return (UIApplicationShortcutIcon*)value;
+    }
+    
     if ([value isKindOfClass:[NSNumber class]]) {
         NSInteger iconIndex = [value integerValue];
         return [UIApplicationShortcutIcon iconWithType:iconIndex];
     }
     
     if ([value isKindOfClass:[NSString class]]) {
-        return [UIApplicationShortcutIcon iconWithTemplateImageName:(NSString*)value];
+        return [UIApplicationShortcutIcon iconWithTemplateImageName:[self urlInAssetCatalog:value]];
     }
     
-    NSLog(@"[ERROR] Invalid icon selection provided, defaulting to SHORTCUT_ICON_TYPE_COMPOSE");
+    NSLog(@"[ERROR] Invalid icon provided, defaulting to SHORTCUT_ICON_TYPE_COMPOSE.");
     return UIApplicationShortcutIconTypeCompose;
 }
 
+-(NSString*)urlInAssetCatalog:(NSString*)url
+{
+    NSString *resultUrl = nil;
+    
+    unsigned char digest[CC_SHA1_DIGEST_LENGTH];
+    NSData *stringBytes = [url dataUsingEncoding: NSUTF8StringEncoding];
+    if (CC_SHA1([stringBytes bytes], (CC_LONG)[stringBytes length], digest)) {
+        // SHA-1 hash has been calculated and stored in 'digest'.
+        NSMutableString *sha = [[NSMutableString alloc] init];
+        for (int i = 0; i < CC_SHA1_DIGEST_LENGTH; i++) {
+            [sha appendFormat:@"%02x", digest[i]];
+        }
+        [sha appendString:@"."];
+        [sha appendString:[url pathExtension]];
+        resultUrl = sha;
```

Comment:
This should be:

```
resultUrl = [NSMutableString stringWithString: sha];
```

else nilling `sha` on the next line will nill `resultUrl` as well.